gossip/comm/comm_impl.go

```diff
@@ -556,7 +564,13 @@ func (c *commImpl) GossipStream(stream proto.Gossip_GossipStreamServer) error {
 	if c.isStopping() {
 		return fmt.Errorf("Shutting down")
 	}
-	connInfo, err := c.authenticateRemotePeer(stream, false)
+	connInfo, err := c.authenticateRemotePeer(stream, false, false)
+
+	if err == errProbe {
```

Comment:
It can be confusing to the code reader. Some questions might arise like why we are ignoring an error. If we are trying to not store info in the conn store for probs, can't the return value be an `isProbe` bool?

Reply:
No, 3 return values is ugly, and this is an un-exported function.
It's perfectly fine to have a specific error that indicates something. 

Look at:

> if e, ok := err.(*QueryError); ok && e.Err == ErrPermission {
    // query failed because of a permission problem
}

From https://blog.golang.org/go1.13-errors.

Reply:
Thanks for the clarification. Agree that 3 return value is ugly.